src/compat/db_json.hpp

```diff
@@ -57,12 +57,36 @@ enum DB_JSON_TYPE
   DB_JSON_BOOL,
 };
 
+class JSON_DOC_STORE
```

Comment:
This concept of let's call it reference store, may be useful for other types of pointers.

I'd like to propose a template implementation that is specialized here for JSON_DOC.

Most operations will be same for all references. In some cases, creating/clearing references may be overwritten.

```
namespace cubmem
{
  template <typename T>
  class pointer_store // or reference_store, or better names
  {
    ...
    void set_as_immutable_reference (T * ptr);
    ...

    void delete_mutable ()  // explicit specializations may do something different
    {
      delete m_mutable_reference;
    }
  };
};
```
```
using JSON_DOC_STORE = cubmem::pointer_store<JSON_DOC>;
void
JSON_DOC_STORE::delete_mutable ()
{
  db_json_delete_doc (m_mutable_reference);
}
```